eth/tracers/internal/tracers/state_diff_tracer.js

```diff
@@ -0,0 +1,537 @@
+		if (!this.hasError && (error !== undefined || opError !== undefined)) {
+			this.hasError = true;
+		}
+
+		if ((error !== undefined || this.includeOpError(opError))
+				&& this.lastAccessedAccount !== null
+				&& this.stateDiff[this.lastAccessedAccount] !== undefined) {
+			this.stateDiff[this.lastAccessedAccount]._error = true;  // mark account that had an error
+			this.lastAccessedAccount = null;
+		}
+	},
+
+	// result is invoked when all the opcodes have been iterated over and returns
+	// the final result of the tracing
+	result: function(ctx, db) {
+		var memoryMarker = this.diffMarkers.Memory;
+
+		// get actual "to" values for from|to|coinbase accounts
+		this.lookupAccount(ctx.from, db);
+		this.lookupAccount(ctx.to, db);
+		this.lookupAccount(ctx.coinbase, db);
+
+		var fromAccountHex = toHex(ctx.from);
+		var toAccountHex = toHex(ctx.to);
+		var coinbaseHex = toHex(ctx.coinbase);
+
+		var convertCtxKeysToBigInt = ['gasLimit', 'gas', 'gasPrice', 'gasUsed', 'value'];
+		for (var i in convertCtxKeysToBigInt) {
+			var key = convertCtxKeysToBigInt[i];
+			ctx[key] = bigInt(ctx[key]);
+		}
+
+		var refund = !this.hasError && this.lastRefund > 0 ? this.lastRefund : 0;
+
+		var gasUsed = ctx.gasLimit.subtract(ctx.gas).add(ctx.gasUsed).add(refund);
+		var gasLeft = ctx.gasLimit.subtract(gasUsed);
+
+		var refundValue = gasLeft.multiply(ctx.gasPrice);
+		var feesValue = gasUsed.multiply(ctx.gasPrice);
+		var fullGasCost = ctx.gasLimit.multiply(ctx.gasPrice);
+
+		// in case from balance is negative because the tracer has disabled the CanTransfer check,
+		// and the Transfer happened before the CaptureStart and the interpreter execution
+		var hasFromSufficientBalanceForValueAndGasCost = ctx.hasFromSufficientBalanceForValueAndGasCost || false;
+		var hasFromSufficientBalanceForGasCost = ctx.hasFromSufficientBalanceForGasCost || false;
+
+		var isCreateType = ctx.type == "CREATE" || ctx.type == "CREATE2";
+		var isCallTypeOnNonExistingAccount = ctx.type == "CALL" && ctx.value.isZero() && !db.exists(ctx.to) && !isPrecompiled(ctx.to)
+
+		// a transaction with "value" set/positive,
+		// while "from" account has not enough balance to pay even for the gas cost,
+		// will not be run at all and will not change the state
+		if (!hasFromSufficientBalanceForGasCost && !hasFromSufficientBalanceForValueAndGasCost && ctx.value.isPositive()) {
+			return {};
+		}
+
+		if (this.stateDiff[fromAccountHex] !== undefined) {
+			var fromAcc = this.stateDiff[fromAccountHex];
+
+			// remove any errors marked on the from account, as it has to be included on output
+			// happens on mordor tx: 0x8f26c1acfce0178a2b037d85feeea99bb961bb46f541ad8c01c6668455952221
+			fromAcc._error = false;
+
+			// don't update from state anymore, as we applied our customs calcs
+			fromAcc._final = true;
+		}
+
+		if (this.stateDiff[toAccountHex] !== undefined) {
+			var toAcc = this.stateDiff[toAccountHex];
+
+			// TODO: check if it needs to be added back
```

Comment:
:tomato: A real pending TODO or DONE?

Reply:
Leftover. Has been tested and it can be removed

Reply:
Resolved by 523cafe